examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython2",
+   "version": "2.7.3"
+  },
+  "name": ""
+ },
+ "nbformat": 3,
+ "nbformat_minor": 0,
+ "worksheets": [
+  {
+   "cells": [
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "# Associative Memory\n",
+      "\n",
+      "This tutorial introduces the Associative Memory (AM) module in the SPA.\n",
+      "\n",
+      "Associative memory is a neural network used to store and recall patterns.\n",
+      "When the network receives a partial or noisy pattern at the input, it can either recover the same pattern or recall another stored pattern. \n",
+      "If the recovered pattern is same as the input pattern, the memory is said to be autoassociative or \"clean-up\" memory. Otherwise, if the recovered pattern is different from the presented one, the network is heteroassociative. \n",
+      "\n",
+      "Patterns stored by the AM module in the SPA are semantic pointers organised in a SPA vocabulary.\n",
+      "The examples in this tutorial demonstrate how to use the AM module, store and recall patterns.\n",
+      "Advanced functionality of the module such as the recall of multiple memories similar to the input are also presented."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# python libraries and plotting tools\n",
+      "import numpy as np\n",
+      "import matplotlib.pyplot as plt\n",
+      "%matplotlib inline\n",
+      "\n",
+      "import nengo\n",
+      "from nengo import spa\n",
+      "%load_ext nengo.ipynb"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 1: Creating an Associative Memory\n",
+      "\n",
+      "We start by creating a set of patterns the AM will store. \n",
+      "The vocabulary in this example contains five words: `ORANGE, APRICOT, CHERRY, STRAWBERRY` and `APPLE`.\n",
+      "Each word is represented as a semantic pointer, a $n$-dimensional vector.\n",
+      "When creating a vocabulary, we specify the number of dimensions for all semantic pointers.\n",
+      "Then, we add the words to a vocabulary with `parse`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": true,
+     "input": [
+      "dim = 32\n",
+      "vocab = spa.Vocabulary(dimensions=dim)\n",
+      "\n",
+      "words = ['ORANGE', 'APRICOT', 'CHERRY', 'STRAWBERRY', 'APPLE']\n",
+      "\n",
+      "for word in words:\n",
+      "    vocab.parse(word)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Next, we create an autoassociative memory with a corresponding set of stored patterns.\n",
+      "The memory is created within a `nengo.Network` module."
```

Comment:
Because the associative memory is a SPA module `spa.SPA` should be used instead of `nengo.Network`. Not sure if using `nengo.Network` might have unintended consequences, but it might lead to confusing problems in user code. Thus, I think we should always use `spa.SPA` when dealing with SPA stuff.